src/probe/probe.c

```diff
@@ -915,6 +917,7 @@ int probe_point_add(uint32_t count, struct probe_point *probe)
 	uint32_t first_free;
 	uint32_t dma_found;
 	struct ipc_comp_dev *dev;
+	struct comp_buffer *caller_id;
```

Comment:
dev can be local within the else {} path from line 956,

s/caller_id/cb

you don't care about dev anywhere, you only need the cb for the `notifier_register`,

Save the cb at line 953: `cb = logging_comp.cb;` instead of dev
Save the cb after line 970: `cb = dev->cb;`

Use the saved `cb` in the `notifier_register()` calls instead of `dev->cb`